gammapy/background/models.py

```diff
@@ -71,3 +81,775 @@ def evaluate(self, x, y):
+            detx_unit = header['TUNIT1']
+        else:
+            raise ValueError("Detector X units not matching ({0}, {1})"
+                             .format(header['TUNIT1'], header['TUNIT2']))
+        if header['TUNIT3'] == header['TUNIT4']:
+            dety_unit = header['TUNIT3']
+        else:
+            raise ValueError("Detector Y units not matching ({0}, {1})"
+                             .format(header['TUNIT3'], header['TUNIT4']))
+        if not detx_unit == dety_unit:
+            ss_error = "This is odd: detector X and Y units not matching"
+            ss_error += "({0}, {1})".format(detx_unit, dety_unit)
+            raise ValueError(ss_error)
+        detx_bins = Angle(detx_bins, detx_unit)
+        dety_bins = Angle(dety_bins, dety_unit)
+
+        # get energy binning
+        energy_bins = _make_bin_edges_array(data['ENERG_LO'], data['ENERG_HI'])
+        if header['TUNIT5'] == header['TUNIT6']:
+            energy_unit = header['TUNIT5']
+        else:
+            raise ValueError("Energy units not matching ({0}, {1})"
+                             .format(header['TUNIT5'], header['TUNIT6']))
+        energy_bins = Quantity(energy_bins, energy_unit)
+
+        # get background data
+        background = data['Bgd'][0]
+        background_unit = header['TUNIT7']
+        TeV_units = ['1/s/TeV/sr', 's-1 sr-1 TeV-1', '1 / (s sr TeV)',
+                     '1 / (TeV s sr)']
+        MeV_units = ['1/s/MeV/sr', 'MeV-1 s-1 sr-1', '1 / (s sr MeV)',
+                     '1 / (MeV s sr)']
+        if background_unit in TeV_units:
+            background_unit = '1 / (s TeV sr)'
+        elif background_unit in MeV_units:
+            background_unit = '1 / (s MeV sr)'
+        else:
+            raise ValueError("Cannot interpret units ({})".format(background_unit))
+        background = Quantity(background, background_unit)
+
+        return CubeBackgroundModel(detx_bins=detx_bins,
+                                   dety_bins=dety_bins,
+                                   energy_bins=energy_bins,
+                                   background=background)
+
+    @staticmethod
+    def from_fits_image(tbhdu):
+        """Read cube background model from a fits image.
+
+        Parameters
+        ----------
+        tbhdu : `~astropy.io.fits.BinTableHDU`
+            HDU binary table for the bg cube
+
+        Returns
+        -------
+        bg_cube : `~gammapy.models.CubeBackgroundModel`
+            bg model cube object
+        """
+        raise NotImplementedError
+
+    @staticmethod
+    def read(filename, format='bin_table'):
+        """Read cube background model from fits file.
+
+        Several input formats are accepted, depending on the value
+        of the 'format' parameter:
+
+        * bin_table (default and preferred format): `~astropy.io.fits.BinTableHDU`
+        * image (alternative format): `~astropy.io.fits.ImageHDU`
+
+        Parameters
+        ----------
+        filename : `~string`
```

Comment:
The Python type is called `str`, not `string`.
Please correct everywhere in the docstrings.